R/dataset.R

```diff
@@ -329,12 +329,39 @@ setMethod("allVariables<-", c("CrunchDataset", "VariableCatalog"),
 setMethod("hidden", "CrunchDataset", function (x) hidden(allVariables(x)))
 
 
-webURL <- function (x) {
+APIToWebURL <- function (x) {
     ## URL to view this dataset in the web app
     stopifnot(is.dataset(x))
     return(paste0(absoluteURL("/", getOption("crunch.api")), "dataset/", id(x)))
 }
 
+webToAPIURL <- function (url) {
+    id <- sub("^https.*?/dataset/([0-9a-f]+)/.*$", "\\1", url)
```

Comment:
Ah right of course. I think we're good. If we run into it again / elsewhere I'll see about generalizing a bit, but for one line here I don't think it's worth holding it up.